lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -710,13 +709,13 @@ class SqlParser(private val ion: IonSystem) : Parser {
             }
             TIME -> {
                 val timeString = token!!.text!!
-                val precision = min(children[0].token?.value?.numberValue()?.toInt() ?: DEFAULT_PRECISION_FOR_TIME, DEFAULT_PRECISION_FOR_TIME)
+                val precision = min(children[0].token?.value?.numberValue()?.toInt() ?: MAX_PRECISION_FOR_TIME, MAX_PRECISION_FOR_TIME)
                 val time = LocalTime.parse(timeString, DateTimeFormatter.ISO_TIME)
                 DateTimeType.Time(time.hour, time.minute, time.second, time.nano, precision, null, metas)
             }
             TIME_WITH_TIME_ZONE -> {
                 val timeString = token!!.text!!
-                val precision = min(children[0].token?.value?.numberValue()?.toInt() ?: DEFAULT_PRECISION_FOR_TIME, DEFAULT_PRECISION_FOR_TIME)
+                val precision = min(children[0].token?.value?.numberValue()?.toInt() ?: MAX_PRECISION_FOR_TIME, MAX_PRECISION_FOR_TIME)
                 val time = OffsetTime.parse(timeString)
                 DateTimeType.Time(time.hour, time.minute, time.second, time.nano, precision, time.offset.totalSeconds/60, metas)
```

Comment:
Some copy/paste that should be avoided--it might be worth capturing a time wrapper of some sort to push this logic outside of the parser or evaluator.

Reply:
Even though this is a little redundant code, it is cleaner this way in my opinion. Moreover, as this code is converting the string token back to the `AST` node , I don't think including this logic in the wrapper class is a great idea.